firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteDocumentOverlayCache.java

```diff
@@ -40,9 +40,11 @@ public SQLiteDocumentOverlayCache(SQLitePersistence db, LocalSerializer serializ
   @Nullable
   @Override
   public Mutation getOverlay(DocumentKey key) {
-    String path = EncodedPath.encode(key.getPath());
-    return db.query("SELECT overlay_mutation FROM document_overlays WHERE uid = ? AND path = ?")
-        .binding(uid, path)
+    String collectionPath = EncodedPath.encode(key.getPath().popLast());
+    String documentId = key.getPath().getLastSegment();
```

Comment:
FYI I just added helpers in https://github.com/firebase/firebase-android-sdk/pull/3173

You can probably just copy the changed from DocumentKey and make them part of this PR as well.

Reply:
I plan to do that in the next PR, the one to release overlay.